src/components/AppHeader/MobileMenu/__tests__/MobileMenu.spec.tsx

```diff
@@ -0,0 +1,85 @@
+import { BrowserRouter } from 'react-router-dom';
+import { QueryParamProvider } from 'use-query-params';
+import { ReactRouter5Adapter } from 'use-query-params/adapters/react-router-5';
+import { useModalManager, useNetworkStatus } from '@/hooks';
+import { useTranslations } from '@deriv-com/translations';
+import { useDevice } from '@deriv-com/ui';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import MobileMenu from '../MobileMenu';
+
+jest.mock('@/hooks', () => ({
+    useModalManager: jest.fn(),
+    useNetworkStatus: jest.fn(),
+    useSyncedTime: jest.fn(),
+}));
+
+jest.mock('@deriv-com/ui', () => ({
+    ...jest.requireActual('@deriv-com/ui'),
+    useDevice: jest.fn(() => ({ isDesktop: false })),
+}));
+
+jest.mock('@deriv-com/api-hooks', () => ({
+    useAuthData: jest.fn().mockReturnValue({
+        isAuthorized: true,
+    }),
+}));
+
+jest.mock('@deriv-com/translations', () => ({
+    useTranslations: jest.fn(),
+}));
+
+const MobileMenuComponent = () => (
+    <BrowserRouter>
+        <QueryParamProvider adapter={ReactRouter5Adapter}>
+            <MobileMenu />
+        </QueryParamProvider>
+    </BrowserRouter>
+);
+
+describe('MobileMenu component', () => {
+    beforeEach(() => {
```

Comment:
is this needed? since its already mocked above

Reply:
Yep nada we need them
at the top I mocked the hooks here I'm mocking the returning value

Reply:
If you have a better suggestion for implementing this please let me know I'll change it

Reply:
so can we not directly mock the return values above itself? will there be a difference?

Reply:
@nada-deriv I will make some changes regarding this 🙏